internal/controller/workloads/rolebasedgroup_controller.go

```diff
@@ -341,6 +370,41 @@ func (r *RoleBasedGroupReconciler) ReconcileScalingAdapter(
 
 	return r.client.Create(ctx, rbgScalingAdapter)
 }
+func (r *RoleBasedGroupReconciler) getOrCreateRevisionIfNotExist(ctx context.Context, rbg *workloadsv1alpha1.RoleBasedGroup) (*appsv1.ControllerRevision, error) {
+	selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			workloadsv1alpha1.SetNameLabelKey: rbg.Name,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	revisions, err := utils.ListRevisions(ctx, r.client, rbg, selector)
```

Comment:
I suggest eliminating the duplicate call to ListRevisions by extracting the listing logic into a single shared helper and letting both NewRevision and getOrCreateRevisionIfNotExist reuse it.